core/matrix/dense.cpp

```diff
@@ -86,6 +99,29 @@ inline void conversion_helper(Csr<ValueType, IndexType> *result,
     exec->run(op(tmp.get(), source));
     tmp->move_to(result);
 }
+
+
+template <typename ValueType, typename IndexType, typename MatrixType,
+          typename OperationType>
+inline void conversion_helper(Ell<ValueType, IndexType> *result,
+                              MatrixType *source, const OperationType &op)
+{
+    auto exec = source->get_executor();
+
+    Array<size_type> max_nonzeros_per_row(exec, 1);
+    exec->run(TemplatedOperation<ValueType>::
+                  make_calculate_max_nonzeros_per_row_operation(
+                      source, max_nonzeros_per_row.get_data()));
+    size_type padding, max_nnz_per_row;
```

Comment:
nit: declare one variable per line, never declare uninitialized variables (it can lead to non-deterministic code and hard to track bugs), and always declare them in the smallest possible scope.

In this case I would do the following:
```c++
const auto max_nnz_per_row = std::max(....);
const auto padding = std::max(...);
```

Reply:
fix it.